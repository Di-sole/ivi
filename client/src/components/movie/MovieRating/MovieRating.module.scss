.rating {
    width: 100%;
    margin-top: 24px;
    pointer-events: none;
    @media (max-width: 1159px) {
        margin-top: 52px;
        margin-bottom: 32px;
    }
    @media (max-width: 879px) {
        margin-top: 24px;
    }
    @media (max-width: 599px) {
        margin-top: 16px;
    }

    display: flex;
    align-items: center;
    /*background-color: rgba(255, 255, 255, 0.08);*/

    &__value {
        display: flex;
        width: 64px;
        min-height: 64px;
        margin-left: -5px;
        height: 28px;

        @media (max-width: 1159px) {
            width: 48px;
            min-height: 48px;
        }
    }

    &__text {
        display: flex;
        flex-direction: column;
        margin-left: 16px;

        font-size: 15px;
        line-height: 20px;
        font-weight: 400;
        color: rgba(255, 255, 255, 0.72);
        text-align: start;

        & > span:first-child {
            font-weight: 600;
            color: #fff;
        }

        & > span:last-child {
            font-size: 13px;
            margin-top: 4px;
            line-height: 16px;
            @media (max-width: 1159px) {
                display: none;
            }
        }
    }

    &__estimate {
        margin-left: auto;
        margin-right: -5px; /*!!!!! 0*/
        border: 1px solid rgba(255, 255, 255, 0.32);
        width: 72px;
        height: 24px;
        border-radius: 6px;
        padding: 3px 7px;

        font-size: 13px;
        line-height: 16px;
        font-weight: 500;
        color: #fff;
    }
}
