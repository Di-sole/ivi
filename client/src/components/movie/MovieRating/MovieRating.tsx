import styles from './MovieRating.module.scss';
import { IMovie } from '../movieMedallion/MovieMedallionsList/Temp/IMovie';
import ButtonUI from '../../UI/buttons/Button/ButtonUI';
import TextSquareUI from '../../UI/squares/TextSquareUI/TextSquareUI';

interface MovieRatingProps {
    movie: IMovie;
}

const MovieRating = ({ movie }: MovieRatingProps) => {
    return (
        <ButtonUI className={styles.rating} shape="large" background="transparent">
            <div className={styles.rating__value}>
                <TextSquareUI value={movie.raiting} textSize="medium" />
            </div>

            <div className={styles.rating__text}>
                <span>Рейтинг</span>
                <span>Интересный сюжет</span>
                <span>118 163 оценки</span>
            </div>

            <div className={styles.rating__estimate}>Оценить</div>
        </ButtonUI>
    );
};

export default MovieRating;
