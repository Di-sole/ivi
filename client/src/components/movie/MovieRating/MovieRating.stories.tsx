import { Meta, StoryObj } from '@storybook/react';
import MovieRating from './MovieRating';
import '../../../styles/nullstyle.scss';
import '../../../styles/globals.scss';
import './MovieRating.module.scss';
import { IMovie } from '../../squareCard/SquareCardsList/Temp/IMovie';

import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';

const meta: Meta<typeof MovieRating> = {
    title: 'Information/Rating',
    component: MovieRating,
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            default: 'desktop',
        },
    },
};

export default meta;

type Story = StoryObj<typeof MovieRating>;

const data: IMovie = {
    id: 1,
    title: 'Название',
    year: 1900,
    ageLimit: '0+',
    time: 'Длительность',
    production: 'Производство',
    genres: ['Жанр', 'Жанр', 'Жанр'],
    displays: ['FullHD', 'HD', '1080', '720'],
    voiceActing: ['Рус', 'Eng'],
    subtitles: ['Рус'],
    subtitlesFull: ['Русский'],
    description: [],
    language: ['Русский', 'Английский'],
    raiting: 9.5,
    nameraiting: 'Рейтинг',
    actors: [],
};

export const FullScreen: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <div style={{ margin: '10px' }}>
                    <Story />
                </div>
            </div>
        ),
    ],
    args: {
        movie: data,
    },
};
