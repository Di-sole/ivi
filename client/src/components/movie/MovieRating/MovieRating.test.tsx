import { render, screen } from '@testing-library/react';
import MovieRating from './MovieRating';
import { movies } from '../movieMedallion/MovieMedallionsList/Temp/Movie.data';

describe('MOVIERATING SNAPSHOTS TESTS', () => {
    test('MovieRating should not be changed', () => {
        render(<MovieRating movie={movies[0]} />);

        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot();
    });
});
