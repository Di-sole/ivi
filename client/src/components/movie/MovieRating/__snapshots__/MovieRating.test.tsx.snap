// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIERATING SNAPSHOTS TESTS MovieRating should not be changed 1`] = `
<button
  class="jsx-3045709936 button rating button"
>
  <div
    class="rating__value"
  >
    <div
      class="value value_medium value_high"
      role="div-text"
    >
      9.5
    </div>
  </div>
  <div
    class="rating__text"
  >
    <span>
      Рейтинг
    </span>
    <span>
      Интересный сюжет
    </span>
    <span>
      118 163 оценки
    </span>
  </div>
  <div
    class="rating__estimate"
  >
    Оценить
  </div>
</button>
`;
